src/utils/openDownloadApp.ts

```diff
@@ -0,0 +1,42 @@
+import { sendAnalyticsEvent } from '@uniswap/analytics'
+import { InterfaceElementName, InterfaceEventName, SharedEventName } from '@uniswap/analytics-events'
+import { isIOS } from 'utils/userAgent'
+
+const APP_STORE_LINK = 'https://apps.apple.com/app/apple-store/id6443944476'
+const MICROSITE_LINK = 'https://wallet.uniswap.org/'
+
+type OpenDownloadAppOptions = {
+  appStoreParams?: string
+  microSiteParams?: string
+  analyticsEventName?: SharedEventName
+}
```

Comment:
IMO this is a case where abstracting the functionality confuses more than it helps. Opening the microsite is very different than opening the app store - can we do two minimal param functions for each of these that log their respective event automatically, no param needed beyond the interface element?

Reply:
@adjkant i think these are coupled somewhat how they are used though, as we basically use the same link to open one or the other in a few different areas, but in one area we need to customize the params. if i split them out then i'd have to add logic to the call-site in multiple places that is duplicated. open to some third alternative or something though

Reply:
@adjkant did a pass over this and i think its a bit cleaner.

Reply:
added the new event as well

Reply:
Looks good on this part!